<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>600</width>
    <height>700</height>
   </rect>
  </property>
  <widget class="QWidget" name="centralwidget">
   <layout class="QVBoxLayout" name="mainVerticalLayout">
    <item>
     <widget class="QGroupBox" name="identityGroup">
      <property name="title">
       <string>User Identity</string>
      </property>
      <layout class="QHBoxLayout" name="identityLayout">
       <item>
        <widget class="QLineEdit" name="nameInput">
         <property name="placeholderText">
          <string>Enter Username...</string>
         </property>
        </widget>
       </item>
      </layout>
     </widget>
    </item>
    <item>
     <widget class="QTabWidget" name="modeTabs">
      <property name="currentIndex">
       <number>1</number>
      </property>
      <widget class="QWidget" name="socketTab">
       <attribute name="title">
        <string>Socket Mode</string>
       </attribute>
       <layout class="QVBoxLayout" name="socketLayout">
        <item>
         <layout class="QHBoxLayout" name="socketConfigLayout">
          <item>
           <widget class="QLineEdit" name="ipInput">
            <property name="text">
             <string>127.0.0.1</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QLineEdit" name="portInput">
            <property name="text">
             <string>2005</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QPushButton" name="connectButton">
            <property name="text">
             <string>Connect</string>
            </property>
           </widget>
          </item>
         </layout>
        </item>
        <item>
         <widget class="QPushButton" name="sendSocketButton">
          <property name="text">
           <string>Send via Network</string>
          </property>
         </widget>
        </item>
       </layout>
      </widget>
      <widget class="QWidget" name="shmTab">
       <attribute name="title">
        <string>Shared Memory</string>
       </attribute>
       <layout class="QVBoxLayout" name="shmLayout">
        <item>
         <layout class="QHBoxLayout" name="shmConfigLayout">
          <item>
           <widget class="QComboBox" name="algoComboBox">
            <item>
             <property name="text">
              <string>FCFS</string>
             </property>
            </item>
            <item>
             <property name="text">
              <string>Priority</string>
             </property>
            </item>
            <item>
             <property name="text">
              <string>Round Robin</string>
             </property>
            </item>
            <item>
             <property name="text">
              <string>SJF</string>
             </property>
            </item>
           </widget>
          </item>
          <item>
           <widget class="QSpinBox" name="prioritySpinBox">
            <property name="minimum">
             <number>1</number>
            </property>
            <property name="maximum">
             <number>5</number>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QPushButton" name="initShmButton">
            <property name="text">
             <string>Initialize SHM</string>
            </property>
           </widget>
          </item>
         </layout>
        </item>
        <item>
         <widget class="QPushButton" name="sendShmButton">
          <property name="text">
           <string>Send to Shared Memory</string>
          </property>
         </widget>
        </item>
       </layout>
      </widget>
     </widget>
    </item>
    <item>
     <widget class="QTextBrowser" name="chatDisplay"/>
    </item>
    <item>
     <widget class="QLineEdit" name="messageInput">
      <property name="placeholderText">
       <string>Type message...</string>
      </property>
     </widget>
    </item>
   </layout>
  </widget>
 </widget>
 <resources/>
 <connections/>
</ui>
